src/backend/joanie/tests/core/test_api_products.py

```diff
@@ -0,0 +1,287 @@
+                                "resource_link": course_run.resource_link,
+                                "state": {
+                                    "priority": course_run.state["priority"],
+                                    "datetime": course_run.state["datetime"]
+                                    .isoformat()
+                                    .replace("+00:00", "Z"),
+                                    "call_to_action": course_run.state[
+                                        "call_to_action"
+                                    ],
+                                    "text": course_run.state["text"],
+                                },
+                                "start": course_run.start.isoformat().replace(
+                                    "+00:00", "Z"
+                                ),
+                                "end": course_run.end.isoformat().replace(
+                                    "+00:00", "Z"
+                                ),
+                                "enrollment_start": course_run.enrollment_start.isoformat().replace(  # noqa pylint: disable=line-too-long
+                                    "+00:00", "Z"
+                                ),
+                                "enrollment_end": course_run.enrollment_end.isoformat().replace(  # noqa pylint: disable=line-too-long
+                                    "+00:00", "Z"
+                                ),
+                            }
+                            for course_run in target_course.course_runs.all().order_by(
+                                "start"
+                            )
+                        ],
+                        "position": target_course.product_relations.get(
+                            product=product
+                        ).position,
+                        "is_graded": target_course.product_relations.get(
+                            product=product
+                        ).is_graded,
+                        "title": target_course.title,
+                    }
+                    for target_course in product.target_courses.all().order_by(
+                        "product_relations__position"
+                    )
+                ],
+                "title": product.title,
+                "type": product.type,
+            },
+        )
+
+    def test_api_product_read_detail_without_course_anonymous(self):
+        """
+        An anonymous user should not be allowed to retrieve a product linked to any course.
+        """
+        product = factories.ProductFactory(courses=[])
+        response = self.client.get(f"/api/products/{product.uid}/")
+
+        self.assertContains(response, "Not found.", status_code=404)
+
+    def test_api_product_read_detail_without_course_authenticated(self):
+        """
+        An authenticated user should not be allowed to retrieve a product linked to any course.
+        """
+        product = factories.ProductFactory(courses=[])
```

Comment:
```suggestion
        product = factories.ProductFactory(courses=[])
```
the link is an empty array, it's weird no? that would happen irl?

Reply:
Yes. In django admin, you create first a product then link this product to a course.